lib/roo_on_rails/context_logging.rb

```diff
@@ -0,0 +1,97 @@
+require 'active_support/core_ext/module/delegation'
+require 'active_support/core_ext/object/blank'
+require 'active_support/version'
+require 'logger'
+require 'roo_on_rails/logfmt'
+
+module RooOnRails
+  # Wraps any standard logger to provide context, similar to `ActiveSupport::TaggedLogging`
+  # but with key/value pairs that are appended to the end of the text.
+  #
+  #   logger = RooOnRails::ContextLogging.new(Logger.new($stdout))
+  #   logger.with(a: 1, b: 2) { logger.info 'Stuff' }                   # Logs "Stuff -- a=1 b=2"
+  #   logger.with(a: 1) { logger.with(b: 2) { logger.info('Stuff') } }  # Logs "Stuff -- a=1 b=2"
+  #
+  # The above methods persist the context in thread local storage so it will be attached to
+  # any logs made within the scope of the block, even in called methods. However, if your
+  # context only applies to the current log then you can chain off the `with` method.
+  #
+  #   logger.with(a: 1, b: 2).info('Stuff')                   # Logs "Stuff -- a=1 b=2"
+  #   logger.with(a: 1) { logger.with(b: 2).info('Stuff')  }  # Logs "Stuff -- a=1 b=2"
+  module ContextLogging
+    module Formatter
+      def call(severity, timestamp, progname, msg)
+        super(severity, timestamp, progname, "#{msg}#{context_text}")
+      end
+
+      def with(**context)
+        current_context.push(context)
+        yield self
+      ensure
+        current_context.pop
+      end
+
+      def clear_context!
+        current_context.clear
+      end
+
+      private
+
+      def current_context
+        # We use our object ID here to avoid conflicting with other instances
+        thread_key = @thread_key ||= "roo_on_rails:logging_context:#{object_id}".freeze
+        Thread.current[thread_key] ||= []
```

Comment:
I'm a bit worried about this. I don't know if the object_id sorts it out.
https://github.com/steveklabnik/request_store

I also think this might be surprising when using promises. 

Would you be able to get the same effect (in rails) by ditching the thread and setting and clearing the log context via a middleware (at the top and the bottom) of the request chain?